### PR TITLE
Help Center: More Guides (in draft) + Centralized Guide Structure

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Adds more guides (in their raw rough draft state - I just copied and pasted the drafts in the doc) and creates a `helpCenterGuideStructure` object that can be used to render the routes and the guides and minimize the areas one needs to update to add/update/remove guides. Also, adds a README with instructions on how to add/update/remove guides. [Once the guides & copy are finalized they should be easy to plug-in and update.]

Note: 
* I haven't implementing linking from the guide to the actual Publisher app - will do that pass in a separate PR that updates the final copy and layout.
* Don't mind the relevant guide links at the moment, I just threw them at random to test the linking between guides - will include those adjustments in the same PR pass as above

Here's a playtesting link incase it's helpful to play around with it in its raw state: 
https://mahmoudtest---justice-counts-web-qqec6jbn6a-uc.a.run.app/help

## Related issues

Contributes to #936

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/HelpCenter/HelpCenterGuides.tsx [hunk 1]
```diff
@@ -16,29 +16,105 @@
 // =============================================================================
 
 import React from "react";
-import { Outlet, useLocation } from "react-router-dom";
+import { Outlet, useLocation, useNavigate } from "react-router-dom";
```
Comment (lilidworkin) [inline publisher/src/components/HelpCenter/HelpCenterGuides.tsx:19]: I wonder if for ease of finding/editing we should have each guide component in its own file? But this is totally up to you

File: publisher/src/components/HelpCenter/HelpCenterSetup.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  DashboardsGuide,
+  ExploreDataGuide,
+  GuideCategories,
+  HelpCenterDirectory,
+  HelpCenterGuideStructure,
+  ManualEntryGuide,
+  SetUpMetricsGuide,
+} from ".";
+
+export const helpCenterGuideStructure: HelpCenterGuideStructure = {
```
Comment (lilidworkin) [inline publisher/src/components/HelpCenter/HelpCenterSetup.tsx:35]: Love this object!

File: publisher/src/components/HelpCenter/README.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
+## How to add or update Help Center Guides
```
Comment (lilidworkin) [inline publisher/src/components/HelpCenter/README.md:1]: I'm OBSESSED with the fact that you took the initiative to add this! Amazing

**Review by lilidworkin — APPROVED** [review]
This looks so cool @mxosman ! I am mostly stamping this, but lmk if there's anything you could use a second pair of eyes on in particular. If you want Jessica/Forrest to sanity check the playtesting link, I might suggest pinging them in Slack since I'm not sure how often they check Github. Looks beautiful and I'm really excited to get this out!

As a heads up we probably will be making changes to this text (since I think both Jessica and Matt are going to take a pass). It's probably tedious to update it, so we might want to distribute that among the team so your eyes don't bleed from copying and pasting for an entire day. Once we have the final final text, you could divy it up and file tasks so everyone gets 1-2 components to update.

**Comment by forrestkim00** [comment]
This looks so so good!! It's always so fun to actually click through the mocks on the actual website. 

Some things we still have to get you: 
- Jessica and I are still deciding on some nav stuff, will get that to you as soon as we finalize it together
- I can provide some examples for how videos/ screenshots will be embedded and provide some guidelines around spacing
- Icons for pages
- LMK if there's anything else that would be helpful to clarify! 

A couple quick nits (writing them down, so I don't forget to mention later): 
1. On the bulk upload page would it be possible to push in the numbers to be in line with the text on top? 
image
2. Very small nit: On the very last line, there's a missing space after "Click Save". 
image

File: publisher/src/components/HelpCenter/HelpCenterGuides.tsx (hunk 1 above)
Comment (mxosman) [reply publisher/src/components/HelpCenter/HelpCenterGuides.tsx:19]: Yeah - that's a really good call! I debated this, but I think your proposal is definitely the best way to organize this. I'll group them all in a folder and include a template file folks can copy and uncomment. Thank you for this!

**Comment by mxosman** [comment]
> [Lili] This looks so cool @mxosman ! I am mostly stamping this, but lmk if there's anything you could use a second pair of eyes on in particular. If you want Jessica/Forrest to sanity check the playtesting link, I might suggest pinging them in Slack since I'm not sure how often they check Github. Looks beautiful and I'm really excited to get this out!
Thank you, Lili! There's nothing in particular that comes to mind - I'm curious to see how this structure works out long term - it feels good so far but always open to feedback as we grow with it! And, thankfully no state is being managed here so that makes things significantly easier to deal with.

> As a heads up we probably will be making changes to this text (since I think both Jessica and Matt are going to take a pass). It's probably tedious to update it, so we might want to distribute that among the team so your eyes don't bleed from copying and pasting for an entire day. Once we have the final final text, you could divy it up and file tasks so everyone gets 1-2 components to update.

Haha - no worries at all! All of that sounds great! I actually don't mind the copying and pasting - I feel like the heavy-lifting is coming up with (and wrangling) the copy and making that uniform (sending prayers to Jessica and Matt).

---

> [Forrest] This looks so so good!! It's always so fun to actually click through the mocks on the actual website.
> 
> Some things we still have to get you:
> 
> * Jessica and I are still deciding on some nav stuff, will get that to you as soon as we finalize it together
> * I can provide some examples for how videos/ screenshots will be embedded and provide some guidelines around spacing
> * Icons for pages
> * LMK if there's anything else that would be helpful to clarify!

Thank you, Forrest! Examples would be super helpful for me to see - thank you. Nothing else comes to mind at the moment - looking forward to seeing what you cook up!

> A couple quick nits (writing them down, so I don't forget to mention later):
> 
> 1. On the bulk upload page would it be possible to push in the numbers to be in line with the text on top?

Absolutely possible! They're just raw `<li></li>` tags for now with no added styling.

> image 
> 2. Very small nit: On the very last line, there's a missing space after "Click Save".

Thank you for pointing this out! I'll do a sweep through of these spacing issues esp. in the final form (they usually come from having an html tag like `<strong>Word</strong>` and how the code formatter breaks the lines after these tags)

---

> [Jessica] +1 to comments above, it's super cool to see everything come together live!!

Thank you, Jessica! Agreed!

> I had a couple thoughts from playtesting through:
> 
> 1. Can we format code differently from text? @forrestkim00 any preferences here?

Absolutely! We can customize it any way we want - I'll style a `<code></code>` HTML tag to look however we want it to look and then all we'll have to do is use that tag wrapped around any code we want to display.

> 2. +1 to Lili's comment on distributing the text updating. I'm happy to own this if that's something you all are comfortable with. I think I can figure out how to change just the text in the code.

I totally feel comfortable with you making updates! I'll make sure the README is all you'll need - and of course I will be here to help along the way.

> 3. (P2) Wondering if we want additional navigation within individual content pages, but this is low priority. Something Forrest and I can think about.

Do you mean like clicking on "Explore Data" in a paragraph in the guide linking directly to the "Explore Data" in the Publisher app?

**Comment by forrestkim00** [comment]
Hey Mahmoud, Jessica and I got a chance to catch up this afternoon! A couple of followups: 
- For the code snippet, this got brought up in our conversation: [prismjs](https://prismjs.com/), up to you to use or not, though!
- For the navigation, I think Jessica was talking about something like this, where a user can jump to different sections on the same page. But I think we're going to not worry about this for now. 
image

**Comment by mxosman** [comment]
> Hey Mahmoud, Jessica and I got a chance to catch up this afternoon! A couple of followups:
> 
> * For the code snippet, this got brought up in our conversation: [prismjs](https://prismjs.com/), up to you to use or not, though!

Ah, nice! Let me see if I can get away with just mirroring their styling since there's only one guide that has code blocks (Automated Bulk Upload) and the code itself doesn't need special syntax highlighting.

> * For the navigation, I think Jessica was talking about something like this, where a user can jump to different sections on the same page. But I think we're going to not worry about this for now.
> 
> image

Ah! Thank you so much for clarifying!

File: publisher/src/components/HelpCenter/HelpCenterGuides.tsx (hunk 1 above)
Comment (mxosman) [reply publisher/src/components/HelpCenter/HelpCenterGuides.tsx]: Feels sooo much better!

**Pull request merged**